src/main/scala/diplomacy/Nodes.scala

```diff
@@ -5,79 +5,238 @@ package freechips.rocketchip.diplomacy
 import Chisel._
 import chisel3.experimental.IO
 import chisel3.internal.sourceinfo.SourceInfo
-import freechips.rocketchip.config.{Parameters,Field}
+import freechips.rocketchip.config.{Field, Parameters}
 import freechips.rocketchip.util.HeterogeneousBag
+
+import scala.collection.immutable
 import scala.collection.mutable.ListBuffer
-import scala.util.matching._
 
+/** A field available in [[Parameters]] used to determine whether [[InwardNodeImp.monitor]] will be called.
+ */
 case object MonitorsEnabled extends Field[Boolean](true)
+
+/** When rendering the edge in graphical format (e.g. GraphML), flip the direction.
+  * yEd tries to put the source node vertically above the sink node. RenderFlipped inverts this.
+  * */
 case object RenderFlipped extends Field[Boolean](false)
 
+/** [[RenderedEdge]] can set the color and label of the visualization of the DAG. */
 case class RenderedEdge(
   colour:  String,
   label:   String  = "",
-  flipped: Boolean = false) // prefer to draw the arrow pointing the opposite direction of other edges
-
-// DI = Downwards flowing Parameters received on the inner side of the node
-// UI = Upwards   flowing Parameters generated by the inner side of the node
-// EI = Edge Parameters describing a connection on the inner side of the node
-// BI = Bundle type used when connecting to the inner side of the node
+  flipped: Boolean = false)
+
+/** [[InwardNodeImp]] defines the types that describe the inward side of the [[BaseNode]].
+  *
+  * @tparam DI The type of the downwards-flowing parameters received on the inner side of the node
+  * @tparam UI The type of the upwards-flowing parameters generated by the inner side of the node
+  * @tparam EI The type of the Edge parameters (after diplomatic resolution) describing a connection on the inner side of the node.
+  * @tparam BI The type of the Bundle used when connecting to the inner side of the node, corresponding to the real hardware interface that is emitted along the graph edge, generally parameterized by the EI type.
+  */
 trait InwardNodeImp[DI, UI, EI, BI <: Data]
 {
+  /** Create an inward-facing edge on this [[BaseNode]]
+    *
+    * @param pd the downward-flowing parameters into the node along the edge
+    * @param pu the upward-flowing parameters going out of the node along the edge
+    * @param p [[Parameters]] which can be accessed by this node as needed
```

Comment:
`a view of [[Parameters]] at the point at which this edge was bound between two nodes`